runtime/src/snapshot_package.rs

```diff
@@ -144,6 +145,12 @@ impl From<AccountsPackage> for SnapshotPackage {
                 accounts_package.archive_format,
             ),
             SnapshotType::IncrementalSnapshot(incremental_snapshot_base_slot) => {
+                snapshot_storages.retain(|storages| {
+                    storages
+                        .first()
+                        .map(|storage| storage.slot() > incremental_snapshot_base_slot)
+                        .unwrap_or_default()
+                });
```

Comment:
The `.first()` here assumes all the storages in this inner Vec all have the same slot, yes?

Also, can you move this:

https://github.com/solana-labs/solana/blob/359e2de090607ded83c1d329ffcf65d2e797ffde/runtime/src/snapshot_package.rs#L81-L86

to just after this `.retain()` call?